website/addons/osfstorage/views.py

```diff
@@ -39,25 +34,28 @@ def make_error(code, message_short=None, message_long=None):
     return HTTPError(code, data=data)
 
 
-def get_record_or_404(path, node_addon):
-    record = model.OsfStorageFileRecord.find_by_path(path, node_addon)
-    if record is not None:
-        return record
-    raise HTTPError(httplib.NOT_FOUND)
-
-
 @must_be_signed
+@utils.handle_odm_errors
 @must_have_addon('osfstorage', 'node')
 def osf_storage_download_file_hook(node_addon, payload, **kwargs):
     try:
-        path = payload['path']
+        path = payload['path'].strip('/')
+        version_id = int(payload.get('version', 0)) - 1
     except KeyError:
-        raise HTTPError(httplib.BAD_REQUEST)
+        raise make_error(httplib.BAD_REQUEST, 'Path is required')
+    except ValueError:
+        raise make_error(httplib.BAD_REQUEST, 'Version must be an int or null')
```

Comment:
This message isn't entirely accurate; if the client passes `null` then `int(None)` will raise a `TypeError`.